File: mockmate-backend/src/main/resources/runners/java/runner_string_array+string_array.java
import java.util.*;
import java.util.regex.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        while (sc.hasNextLine()) sb.append(sc.nextLine()).append(" ");
        String fullInput = sb.toString();

        try {
            List<String[]> arrays = new ArrayList<>();
            Matcher mBracket = Pattern.compile("\\[(.*?)\\]").matcher(fullInput);
            while(mBracket.find()) {
                String arrayPart = mBracket.group(1);
                List<String> list = new ArrayList<>();
                Matcher m = Pattern.compile("\"([^"]*)\"|'([^']*)'|([^\\s,]+)").matcher(arrayPart);
                while(m.find()) {
                    if(m.group(1) != null) list.add(m.group(1));
                    else if(m.group(2) != null) list.add(m.group(2));
                    else list.add(m.group(3));
                }
                arrays.add(list.toArray(new String[0]));
            }
            String[] arr1 = arrays.size() > 0 ? arrays.get(0) : new String[0];
            String[] arr2 = arrays.size() > 1 ? arrays.get(1) : new String[0];

            Solution sol = new Solution();
            Object result = sol.{{methodSignature}}(arr1, arr2);
            if(result instanceof String[]) System.out.println(Arrays.toString((String[])result));
            else System.out.println(result);
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        } finally {
            sc.close();
        }
    }
}

{{USER_CODE}}

File: mockmate-backend/src/main/resources/runners/python/runner_string_array+string_array.py
import sys
import json
import re

def main():
    input_data = sys.stdin.read().strip()
    if not input_data:
        return

    arrays = []
    # Find all top level arrays
    depth = 0
    start = -1
    for i, c in enumerate(input_data):
        if c == '[':
            if depth == 0: start = i
            depth += 1
        elif c == ']':
            depth -= 1
            if depth == 0 and start != -1:
                try:
                    arrays.append(json.loads(input_data[start:i+1].replace("'", '"')))
                except:
                    pass
                start = -1

    arr1 = arrays[0] if len(arrays) > 0 else []
    arr2 = arrays[1] if len(arrays) > 1 else []

    sol = Solution()
    result = sol.{{methodSignature}}(arr1, arr2)

    if isinstance(result, bool):
        print(str(result).lower())
    else:
        print(json.dumps(result).replace(' ', ''))

{{USER_CODE}}

if __name__ == '__main__':
    main()
